### PR TITLE
Add support for list arguments with evtset.rename(...)

### PR DESCRIPTION
**Pull request by achoum**

### CODE REVIEW

File: temporian/core/event_set_ops.py [hunk 1]
```diff
@@ -3381,19 +3381,20 @@ def propagate(
...
-        If the input has multiple features, then `features` must be a mapping
-        with the old names as keys and the new names as values.
+        If the input has multiple features, then `features` can either be (1) a
+        dictionary mapping old names to the new names, or (2) a list of new
+        names of the same size as `evtset.schema.feature_names()`.
 
-        The indexes renaming follows the same criteria, accepting a single string or
-        a mapping for multiple indexes.
+        The indexes renaming follows the same criteria, accepting a single
+        string or a mapping for multiple indexes.
```
Comment (ianspektor) [inline temporian/core/event_set_ops.py:3397]: missing the list option here

File: temporian/core/operators/rename.py [hunk 2]
```diff
@@ -123,6 +123,22 @@ def _normalize_rename_features(
                 "of rename strings instead."
             )
         features = {schema.features[0].name: features}
+    elif isinstance(features, list):
+        if len(features) != len(schema.features):
+            raise ValueError(
+                f"The provided {len(features)} new names ({features!r}) don't"
+                f" match the number of features ({len(schema.features)}) in the"
+                " event-set."
```
Comment (ianspektor) [inline temporian/core/operators/rename.py:131]: s/event-set/EventSet

File: temporian/core/operators/rename.py [hunk 3]
```diff
@@ -152,6 +168,22 @@ def _normalize_rename_indexes(
                 "of rename strings instead."
             )
         indexes = {schema.indexes[0].name: indexes}
+    elif isinstance(indexes, list):
+        if len(indexes) != len(schema.indexes):
+            raise ValueError(
+                f"The provided {len(indexes)} new names ({indexes!r}) don't"
+                f" match the number of indexes ({len(schema.indexes)}) in the"
+                " event-set."
```
Comment (ianspektor) [inline temporian/core/operators/rename.py:176]: s/event-set/EventSet

**Review by ianspektor — APPROVED** [review]
👍🏼👍🏼👍🏼 thanks!

File: temporian/core/event_set_ops.py (hunk 1 above)
Comment (achoum) [reply temporian/core/event_set_ops.py:3397]: Good catch.

**Pull request merged**